zapr_test.go

```diff
@@ -0,0 +1,376 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+Copyright 2021 The logr Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package zapr_test
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/zapr"
+)
+
+const fixedTime = 123.456789
+
+func fixedTimeEncoder(_ time.Time, enc zapcore.PrimitiveArrayEncoder) {
+	enc.AppendFloat64(fixedTime)
+}
+
+func newZapLogger(lvl zapcore.Level, w zapcore.WriteSyncer) *zap.Logger {
+	if w == nil {
+		w = zapcore.AddSync(io.Discard)
+	}
+	encoder := zapcore.NewJSONEncoder(zapcore.EncoderConfig{
+		MessageKey:     "msg",
+		CallerKey:      "caller",
+		TimeKey:        "ts",
+		EncodeTime:     fixedTimeEncoder,
+		EncodeDuration: zapcore.StringDurationEncoder,
+		EncodeCaller:   zapcore.ShortCallerEncoder,
+	})
+	core := zapcore.NewCore(encoder, zapcore.AddSync(w), lvl)
+	l := zap.New(core, zap.WithCaller(true))
+	return l
+}
+
+// TestInfo tests the JSON info format.
+func TestInfo(t *testing.T) {
```

Comment:
This is a pretty complex test.  I won't demand it now, since I know it is ported, but it would be nice to have 1 test that proves Info() works with no options.  Then a test that proves that LogInfoLevel works.  Then a test that proves ErrorKey works.  etc.  I feel like each test should be much simpler to comprehend